### PR TITLE
Add bindings_perl and examples/perl_api_example

### PR DESCRIPTION
**Pull request by xanni**

### CODE REVIEW

File: examples/perl_api_example/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+
+## Python API Example
```
Comment (mlangkabel) [inline examples/perl_api_example/README.md:2]: I guess you mean Perl here ;)

File: examples/perl_api_example/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+* CMake
+* C++ Compiler
+* Perl
+* SWIG
+
+### Running from Command Line
+
+* Run cmake with `BUILD_BINDINGS_PERL=ON`
+* Build target `sourcetraildb`
```
Comment (mlangkabel) [inline examples/perl_api_example/README.md:14]: I think it's called "bindings_perl", right? It is defined like that in the main CMakeLists.txt file

File: examples/perl_api_example/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+* Run example from the directory `SourcetrailDB/examples/perl_api_example/src` with:
+
+```
+$ perl example.pl --database-file-path=absolute/path/to/SourcetrailDB/examples/perl_api_example/src/example.srctrldb --source-file-path=absolute/path/to/SourcetrailDB/examples/perl_api_example/data/file.pm
+```
+
+### Running with Sourcetrail
+
+* Run cmake with `BUILD_EXAMPLES=ON` and `BUILD_BINDINGS_PERL=ON`
+* Build target `sourcetraildb` and `perl_api_example`.
```
Comment (mlangkabel) [inline examples/perl_api_example/README.md:29]: "bindings_perl" again?

File: examples/perl_api_example/README.md (hunk 1 above)
Comment (xanni) [reply examples/perl_api_example/README.md:2]: Oops!  Corrected.

File: examples/perl_api_example/README.md (hunk 2 above)
Comment (xanni) [reply examples/perl_api_example/README.md:14]: You're right.  This was also corespondingly incorrect in the python_api_example - both corrected.

File: examples/perl_api_example/README.md (hunk 3 above)
Comment (xanni) [reply examples/perl_api_example/README.md:29]: Yes.  Again, corrected in both perl_api_example and python_api_example

**Comment by mlangkabel** [comment]
@xanni: Thanks for resolving! I'm currently trying to get this PR up and running on Windows, but currently I have some issues with the Perl install. Somehow it doesn't come with a `libperl.lib`. Do you have any experience with that?

**Comment by xanni** [comment]
> @xanni: Thanks for resolving! I'm currently trying to get this PR up and running on Windows, but currently I have some issues with the Perl install. Somehow it doesn't come with a `libperl.lib`. Do you have any experience with that?

Unfortunately I don't currently have a Windows development system and I haven't touched Perl on Windows for years.  I'll see what I can do.

**Comment by zmughal** [comment]
This will need to be rebased on master (to fix the SWIG install on AppVeyor) and the Perl bindings needed to be added to both Travis CI and AppVeyor CI configs. I have experience with using Travis CI and AppVeyor with Perl builds, so I can help with that.

I have also created a possible PR on top of this at <https://github.com/xanni/SourcetrailDB/pull/1>.

**Comment by mlangkabel** [comment]
I decided to merge this PR (in a rebased version #34), even though this does not (yet) have any CI on Windows, because I don't want all this work to fall too far behind once we continue working on SourcetrailDB. 

So thank you again for working on this @xanni!

@zmughal: Thanks for offering your help! You can now fork this repo directly and create a pull request for making this work on windows.

**Comment by xanni** [comment]
Thanks!  Glad to hear you're able to make use of it and sorry I haven't had a chance to try and get it working on Windows.

**Pull request closed without merging**